packages/google_maps_flutter/google_maps_flutter_android/android/src/test/java/io/flutter/plugins/googlemaps/HeatmapsControllerTest.java

```diff
@@ -0,0 +1,108 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.googlemaps;
+
+import static io.flutter.plugins.googlemaps.Convert.HEATMAP_DATA_KEY;
+import static io.flutter.plugins.googlemaps.Convert.HEATMAP_ID_KEY;
+import static io.flutter.plugins.googlemaps.Convert.HEATMAP_OPACITY_KEY;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+import android.os.Build;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.TileOverlay;
+import com.google.android.gms.maps.model.TileOverlayOptions;
+import com.google.maps.android.heatmaps.HeatmapTileProvider;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = Build.VERSION_CODES.P)
```

Comment:
For my own knowledge does this need to be run on only p or is P the floor api where this test will work?

Reply:
In that case can you set minSdk=Build.VERSION_CODES.LOLLIPOP 
If that gives you trouble for some reason you can leave it P but I would appreciate if you set minSdk instead of sdk. The difference is it runs the unit tests against every version after that value. Helping us catch issues when new android apis come out.

Reply:
Should we create an issue to update the other tests?

Reply:
I will take care of it as part of an issue i am writing for another engineer. Good idea.